src/Aquifer.Public.API/Endpoints/Resources/Updates/List/Validator.cs

```diff
@@ -9,5 +11,12 @@ public Validator()
     {
         RuleFor(x => x.Timestamp).NotEmpty().GreaterThanOrEqualTo(DateTime.UtcNow.AddDays(-90).Date);
         RuleFor(x => x.LanguageId).InclusiveBetween(1, 32);
+        RuleFor(x => x.Offset).GreaterThanOrEqualTo(0);
+        RuleFor(x => x.Limit).InclusiveBetween(0, 100);
+        RuleFor(x => x)
+            .Must(x => (x.LanguageCode is not null && x.LanguageId == default)
+                       || (x is { LanguageId: not null, LanguageCode: null })
+                       || (x.LanguageId == default && x.LanguageCode is null))
+            .WithMessage("Either languageId or languageCode is optional. Cannot use both.");
```

Comment:
This message is a little confusing. I would just say "Cannot use both languageId and languageCode".

Also, I would be careful about using `default`, because it can cause unexpected behavior sometimes. In this case `default` is null, but if you changed `LanguageId` to `int`, then the default would be `0`. So in this situation it may not matter, but you could potentially create an unexpected error when changing the nullability of a type when you're using `default`.